### DISCUSSION TITLE
FR: Visual aid for Entry Type and Block Type selection

### DISCUSSION THREAD

**Discussion by Mosnar** [craftcms/cms#8656] [Ideas] [answered]
### Description
One of the most ubiquitous uses for entry types is for page template selection. Often we have entry types laid out like this:
- Listing Page
- Content Detail Page
- Some variant of a detail page
- Product Landing Page
- Product Detail Page

Content authors and website owners often have trouble visualizing how these page templates might be used and how they look. I think it would be beneficial to introduce some sort of flexible visual aid, such as the ability to pick an icon to represent an entry type.

These could be smaller icons, such as this screenshot from Slickplan:
![image](https://user-images.githubusercontent.com/3810939/55328294-076ec300-545a-11e9-8727-e8b83acbf289.png)

Or larger images, such as this screenshot from Flowmapp
![image](https://user-images.githubusercontent.com/3810939/55328411-4d2b8b80-545a-11e9-9763-8868becfa257.png)

I'm imagining this UI may also be applicable to Matrix Blocks to assist with Page Builders & Content Builders.

### Additional info

- Craft version: 3.1.x

**Comment by brandonkelly** [comment]
Block type icons were already requested by #2810, but I’ll leave this open for entry types.

**Comment by aarmitage** [comment]
I would add my support to something like this too. The current placement of the entry type dropdown is out of the natural visual flow when creating entries with type options. The visual aid here would be a great enhancement, but also the more obvious choice of entry types for a given section rather than it being out of sight in the right hand panel.

**Comment by brandonkelly** [comment, marked as answer]
Entry types will be able to have icons in Craft 5 (#14169) 🎉

**Reply by brandonkelly** [reply]
Craft 5.0.0-alpha.7 is out now with entry type icons.
